src/main/scala/diplomacy/Clone.scala

```diff
@@ -17,5 +19,21 @@ final class CloneLazyModule private (val base: LazyModule)
 
 object CloneLazyModule
 {
+  /** The old API **/
   def apply(base: LazyModule) = new CloneLazyModule(base)
+
+
+  /** Constructs a [[LazyModule]], but replaces its [[LazyModuleImp]] with a cloned [[LazyModuleImp]]
```

Comment:
What will happen if `LazyModule` to be cloned contains any sub-`LazyModule`:
1. What will happen for the  forwarded dangle
2. What if the submodule in `bc` is not same with `cloneProto`(can we check it)?

Reply:
1. The dangles of the child lazy modules of `bc` are elaborated, but no hw is bound to them. This still checks sanity in `bc` (any diplomacy errors in lazy module elaboration of `bc` would still appear if you use `CloneLazyModule(bc, ...)` instead of `LazyModule(bc)`.).

2. I can add a test that the child classes are equivalent. Ideally, we should be able to check that the diplomacy parameters for those classes are equivalent too, but that is not possible right now.